Add MCP analytics product landing page (#17647)
PostHog/posthog.com: contents/docs/mcp-analytics/index.mdx

<CalloutBox icon="IconInfo" title="MCP analytics is in alpha" type="fyi">

> `@posthog/mcp` is a TypeScript SDK for instrumenting Model Context Protocol (MCP) servers with PostHog analytics. It's published on npm as a `0.1.x` release while we build it in public. The API, event names, properties, and tracing behavior may change before the SDK reaches `1.0`. Don't depend on it for production reporting yet. Today, only TypeScript MCP servers are supported – a Python SDK is on the roadmap.

Reviewer: Version numbers in prose go stale fast (npm's already on `0.4.0`, not `0.1.x`), so I'd just drop the specific version instead of chasing it. Same idea in `start-here.mdx`.

```suggestion
`@posthog/mcp` is a TypeScript SDK for instrumenting Model Context Protocol (MCP) servers with PostHog analytics. It's published on npm and we're building it in public, so the API, event names, properties, and tracing behavior may change before the SDK reaches `1.0`. Don't depend on it for production reporting yet. Today, only TypeScript MCP servers are supported – a Python SDK is on the roadmap.
```

</CalloutBox>

MCP analytics helps you understand how AI agents actually use the MCP server you ship: which tools get called, how often, what intent the agent had, where calls are failing, what tools the agent asked for but you don't offer yet, and how individual sessions flow end to end.

> The SDK wraps your existing MCP server and emits PostHog events on every tool call, resource read, prompt fetch, and initialize handshake. You can keep using whatever tooling you already have on top of PostHog – insights, dashboards, alerts, Error Tracking – without any additional ingestion plumbing.

Reviewer: `instrument()` only auto-wraps `tools/call`, `tools/list`, and `initialize`. The resource/prompt event constants exist but nothing fires them yet, so "resource read, prompt fetch" overclaims what you get today. Trimmed it.

```suggestion
The SDK wraps your existing MCP server and emits PostHog events on every tool call, `tools/list`, and initialize handshake. You can keep using whatever tooling you already have on top of PostHog – insights, dashboards, alerts, Error Tracking – without any additional ingestion plumbing.
```
